### PR TITLE
Add importable test framework for context menu plugins

### PR DESCRIPTION
**Pull request by grant-cleary**
Similar to the test framework we've added for plugin classes, we also want to add one for context menu plugins.

### CODE REVIEW

File: components/context-menu/formatting/formatting.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const disallowedNodes = new Set(['IMG', 'FIGURE', 'AUDIO', 'VIDEO', 'IFRAME', 'O
 
 export class FormattingContextMenuPlugin {
 
-	static get name() {
+	static get pluginName() {
```
Comment (grant-cleary) [inline components/context-menu/formatting/formatting.js:16]: The test exposed something funny. about this being a static property... Specifically, `name` is already an implicit property on any class. So if this wasn't defined within this plugin, `FormattingContextMenuPlugin.name` would still return `FormattingContextMenuPlugin`. This is not ideal and is definitely not intended, so I'm switching the property name to `pluginName` to force it to be explicitly set.

File: components/context-menu/link/link-edit.js [hunk 2]
```diff
@@ -13,11 +13,11 @@ const elementName = 'd2l-htmleditor-link-edit-context-menu';
 
 export class LinkEditContextMenuPlugin {
 
-	static get name() {
+	static get pluginName() {
 		return 'link-edit';
 	}
 
-	static get tagName() {
```
Comment (grant-cleary) [inline components/context-menu/link/link-edit.js:20]: I don't know why this is static. The functionality requiring use of the tag name isn't needed for this particular context menu, which is the only reason this isn't broken in production.

**Review by JakeProuse — APPROVED** [review]

File: test/context-menu-plugins.test.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		{ name: 'insert-element', plugin: InsertElementContextMenuPlugin },
+		{ name: 'layouts', plugin: LayoutsContextMenuPlugin },
+		{ name: 'practices', plugin: PracticesContextMenuPlugin }
+	].forEach(info => {
+		describe(`${info.name} plugin structure`, () => {
+			it('should pass validation', () => {
+				const result = verifyContextMenuPluginStructure(info.plugin);
+				expect(
+					result.success,
+					'Context menu plugin verification failed. Encountered the following errors:\n\n'
```
Comment (PaulKrznaric) [inline test/context-menu-plugins.test.js:25]: Doesn't this seem contradictory? Validation failed, but we're expecting a success/it to pass validation?

**Review by PaulKrznaric — APPROVED** [review]

File: test/context-menu-plugins.test.js (hunk 3 above)
Comment (grant-cleary) [reply test/context-menu-plugins.test.js:25]: The second parameter passed to `expect` is a message that's meant to be displayed on failure.

**Pull request merged**